File: app/src/main/java/com/soutvoid/gamesproject/interactor/util/RealmWrapUtils.java
package com.soutvoid.gamesproject.interactor.util;

import com.annimon.stream.Collectors;
import com.annimon.stream.Stream;

import java.util.List;

import io.realm.RealmList;
import io.realm.RealmObject;

public class RealmWrapUtils {

    public static <T extends RealmObject> RealmList<T> convertList(List<T> original) {
        RealmList<T> result = new RealmList<>();
        result.addAll(original);
        return result;
    }

    public static RealmString wrapString(String string) {
        return new RealmString(string);
    }

    public static RealmLong wrapLong(long val) {
        return new RealmLong(val);
    }

    public static RealmList<RealmString> wrapStrings(List<String> strings) {
        return RealmWrapUtils.convertList(
                Stream.of(strings)
                        .map(RealmWrapUtils::wrapString)
                        .collect(Collectors.toList()));
    }

    public static RealmList<RealmLong> wrapLongs(List<Long> Longs) {
        return RealmWrapUtils.convertList(
                Stream.of(Longs)
                        .map(RealmWrapUtils::wrapLong)
                        .collect(Collectors.toList()));
    }

}

File: app/src/main/java/com/soutvoid/gamesproject/interactor/util/RealmWrapUtils.kt
package com.soutvoid.gamesproject.interactor.util

import io.realm.RealmList
import io.realm.RealmObject

object RealmWrapUtils {

    fun <T : RealmObject> convertList(original: List<T>): RealmList<T> {
        val result = RealmList<T>()
        result.addAll(original)
        return result
    }

    fun wrapString(string: String): RealmString {
        return RealmString(string)
    }

    fun wrapLong(`val`: Long): RealmLong {
        return RealmLong(`val`)
    }

    fun wrapStrings(strings: List<String>): RealmList<RealmString> {
        return RealmWrapUtils.convertList(
                strings.mapIndexed { _, s -> wrapString(s) }.toList())
    }

    fun wrapLongs(Longs: List<Long>): RealmList<RealmLong> {
        return RealmWrapUtils.convertList(
                Longs.map { wrapLong(it) }.toList())
    }

}
